taskweaver/llm/__init__.py

```diff
@@ -81,6 +83,8 @@ def __init__(
             self.embedding_service = PlaceholderEmbeddingService(
                 "Azure ML does not support embeddings yet. Please configure a different embedding API.",
             )
+        elif self.config.embedding_api_type == "groq":
+            self._set_embedding_service(GroqService)
```

Comment:
Does groq have embedding function?

Reply:
Groq does not natively offer its embedding function. But we can use the embeddings model of hugging face or from transformers directly.

For example, we can use the following structure for embeddings:
```
from transformers import AutoModel

embedding_model = AutoModel.from_pretrained('jinaai/jina-embeddings-v2-base-en', trust_remote_code=True)
user_query = "user query"
query_embeddings = `embedding_model.encode(user_query).tolist()
```

I ran some queries which worked, so I thought I could skip embeddings.
Please let me know if any changes or updates required😊